### PR TITLE
Fix flake8 errors

### PR DESCRIPTION
**Pull request by pnbabu**
- Add error and warning codes to flake8 `--ignore` list
- Fix the errors raised by flake8 on `.py` files
- Remove `include_checker.py` file as it is no longer used

### CODE REVIEW

File: pynest/nest/lib/hl_api_helper.py [hunk 1]
```diff
@@ -159,22 +152,7 @@ def new_func(*args, **kwargs):
...
-
-    """
-    import sys
-    if sys.version_info[0] < 3:
-        return basestring
-    return str
-
-
-uni_str = get_unistring_type()
+uni_str = str
```
Comment (terhorstd) [inline pynest/nest/lib/hl_api_helper.py]: do we need both?

File: .github/workflows/nestbuildmatrix.yml [hunk 2]
```diff
@@ -198,7 +178,7 @@ jobs:
 
       - name: "Run flake8..."
         run: |
-          flake8 pynest
+          flake8 pynest/
```
Comment (jougs) [inline .github/workflows/nestbuildmatrix.yml]: Shouldn't we also run this on the tests in `testsuite`?

File: pynest/nest/__init__.py [hunk 3]
```diff
@@ -81,27 +81,27 @@ def __init__(self, name):
...
-        _rel_import_star(self, ".lib.hl_api_types")
+        _rel_import_star(self, ".lib.hl_api_connections")           # noqa: F821
+        _rel_import_star(self, ".lib.hl_api_exceptions")            # noqa: F821
+        _rel_import_star(self, ".lib.hl_api_info")                  # noqa: F821
+        _rel_import_star(self, ".lib.hl_api_models")                # noqa: F821
+        _rel_import_star(self, ".lib.hl_api_nodes")                 # noqa: F821
+        _rel_import_star(self, ".lib.hl_api_parallel_computing")    # noqa: F821
+        _rel_import_star(self, ".lib.hl_api_simulation")            # noqa: F821
+        _rel_import_star(self, ".lib.hl_api_spatial")               # noqa: F821
+        _rel_import_star(self, ".lib.hl_api_types")                 # noqa: F821
```
Comment (jougs) [inline pynest/nest/__init__.py:95]: :heart:

File: pynest/nest/__init__.py [hunk 4]
```diff
@@ -506,4 +506,7 @@ def lazy_loader(self):
 globals().update(_module.__dict__)
 
 # Clean up obsolete references
+# Since these references are deleted, flake8 complains with an error
+# `F821 undefined name` where these variables are used. Hence we mark all these
```
Comment (jougs) [inline pynest/nest/__init__.py]:
```suggestion
# `F821 undefined name` where these variables are used. Hence we mark all those
```

File: .github/workflows/nestbuildmatrix.yml (hunk 2 above)
Comment (pnbabu) [reply .github/workflows/nestbuildmatrix.yml]: Yes, there are also some `.py` files outside of `pynest` and `testsuite` directories. So I think it is better to run flake8 on the entire nest repository.

File: pynest/nest/lib/hl_api_spatial.py [hunk 5]
```diff
@@ -959,8 +958,6 @@ def PlotLayer(layer, fig=None, nodecolor='b', nodesize=20):
 
     elif len(ext) == 3:
         # 3D layer
```
Comment (jougs) [inline pynest/nest/lib/hl_api_spatial.py]:
```suggestion
```

File: pynest/nest/lib/hl_api_spatial.py [hunk 6]
```diff
@@ -1100,8 +1097,6 @@ def PlotTargets(src_nrn, tgt_layer, syn_type=None, fig=None,
 
     else:
         # 3D layer
```
Comment (jougs) [inline pynest/nest/lib/hl_api_spatial.py]:
```suggestion
```

**Review by jougs — CHANGES_REQUESTED** [review]
Just two more minor removals and we're good to merge from my side

**Comment by pnbabu** [comment]
> Just two more minor removals and we're good to merge from my side

After running flake8 on the entire repository, there are more errors to be fixed. I think we should merge this after fixing them.

File: pynest/nest/__init__.py (hunk 3 above)
Comment (terhorstd) [reply pynest/nest/__init__.py:95]: :rocket:

**Review by terhorstd — APPROVED** [review]
Thanks @pnbabu ! That is already a lot of fixing you added here! :rocket: 

LGTM

**Pull request merged**